### PR TITLE
move advective boundary conditions into divergence operator

### PR DESCRIPTION
**Pull request by simonbyrne**
## Purpose 
Alternative fix to #1611.

## To-do
@valeriabarra can you apply the same tests as in #1612?

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/tendencies/implicit/implicit_tendency.jl [hunk 1]
```diff
@@ -9,28 +9,24 @@ import ClimaCore: Fields
 # TODO: Can we rewrite ᶠfct_boris_book and ᶠfct_zalesak so that their broadcast
 # expressions are less convoluted?
 vertical_transport!(ᶜρχₜ, ᶜJ, ᶜρ, ᶠu³, ᶜχ, dt, ::Val{:none}) =
-    @. ᶜρχₜ = -(ᶜdivᵥ(ᶠwinterp(ᶜJ, ᶜρ) * ᶠu³ * ᶠinterp(ᶜχ)))
+    @. ᶜρχₜ = -(ᶜadvdivᵥ(ᶠwinterp(ᶜJ, ᶜρ) * ᶠu³ * ᶠinterp(ᶜχ)))
```
Comment (dennisYatunin) [inline src/tendencies/implicit/implicit_tendency.jl:12]: I think this one can be left as `ᶜdivᵥ`. If this does get changed, it might be good to also change the corresponding Jacobian term on line 32 of `wfact.jl`.

File: src/tendencies/implicit/implicit_tendency.jl [hunk 2]
```diff
@@ -9,28 +9,24 @@ import ClimaCore: Fields
...
+    @. ᶜρχₜ = -(ᶜadvdivᵥ(ᶠwinterp(ᶜJ, ᶜρ) * ᶠupwind3(ᶠu³, ᶜχ)))
 vertical_transport!(ᶜρχₜ, ᶜJ, ᶜρ, ᶠu³, ᶜχ, dt, ::Val{:boris_book}) =
-    @. ᶜρχₜ = -(ᶜdivᵥ(
-        ᶠwinterp(ᶜJ, ᶜρ) * ᶠset_upwind_bcs(
-            ᶠupwind1(ᶠu³, ᶜχ) + ᶠfct_boris_book(
-                ᶠupwind3(ᶠu³, ᶜχ) - ᶠupwind1(ᶠu³, ᶜχ),
-                ᶜχ / dt - ᶜdivᵥ(ᶠwinterp(ᶜJ, ᶜρ) * ᶠupwind1(ᶠu³, ᶜχ)) / ᶜρ,
-            ),
+    @. ᶜρχₜ = -(ᶜadvdivᵥ(
+        ᶠwinterp(ᶜJ, ᶜρ) * ᶠupwind1(ᶠu³, ᶜχ) + ᶠfct_boris_book(
```
Comment (dennisYatunin) [inline src/tendencies/implicit/implicit_tendency.jl:19]:
```suggestion
        ᶠwinterp(ᶜJ, ᶜρ) * (
            ᶠupwind1(ᶠu³, ᶜχ) + ᶠfct_boris_book(
                 ᶠupwind3(ᶠu³, ᶜχ) - ᶠupwind1(ᶠu³, ᶜχ),
                 ᶜχ / dt - ᶜdivᵥ(ᶠwinterp(ᶜJ, ᶜρ) * ᶠupwind1(ᶠu³, ᶜχ)) / ᶜρ,
             ),
         ),
```

File: src/tendencies/implicit/implicit_tendency.jl [hunk 3]
```diff
@@ -9,28 +9,24 @@ import ClimaCore: Fields
...
 vertical_transport!(ᶜρχₜ, ᶜJ, ᶜρ, ᶠu³, ᶜχ, dt, ::Val{:zalesak}) =
-    @. ᶜρχₜ = -(ᶜdivᵥ(
-        ᶠwinterp(ᶜJ, ᶜρ) * ᶠset_upwind_bcs(
-            ᶠupwind1(ᶠu³, ᶜχ) + ᶠfct_zalesak(
-                ᶠupwind3(ᶠu³, ᶜχ) - ᶠupwind1(ᶠu³, ᶜχ),
-                ᶜχ / dt,
-                ᶜχ / dt - ᶜdivᵥ(ᶠwinterp(ᶜJ, ᶜρ) * ᶠupwind1(ᶠu³, ᶜχ)) / ᶜρ,
-            ),
+    @. ᶜρχₜ = -(ᶜadvdivᵥ(
+        ᶠwinterp(ᶜJ, ᶜρ) * ᶠupwind1(ᶠu³, ᶜχ) + ᶠfct_zalesak(
```
Comment (dennisYatunin) [inline src/tendencies/implicit/implicit_tendency.jl:26]:
```suggestion
        ᶠwinterp(ᶜJ, ᶜρ) * (
            ᶠupwind1(ᶠu³, ᶜχ) + ᶠfct_zalesak(
                 ᶠupwind3(ᶠu³, ᶜχ) - ᶠupwind1(ᶠu³, ᶜχ),
                 ᶜχ / dt,
                 ᶜχ / dt - ᶜdivᵥ(ᶠwinterp(ᶜJ, ᶜρ) * ᶠupwind1(ᶠu³, ᶜχ)) / ᶜρ,
             ),
         ),
```

File: src/tendencies/implicit/implicit_tendency.jl (hunk 1 above)
Comment (simonbyrne) [reply src/tendencies/implicit/implicit_tendency.jl:12]: Okay. I wanted to keep them all consistent.

**Comment by valeriabarra** [comment]
Tests look better with this version. Here is the `v` component, at level = 1, day 10 of simulation for the moist BW test case with `zalesak` on tracers:
v_day10_level1

I would merge this PR with the fix

**Review by valeriabarra — APPROVED** [review]
